src/JabbR-Core/Commands/CommandManager.cs

```diff
@@ -0,0 +1,214 @@
+                              IChatService service,
+                              IJabbrRepository repository,
+                              ICache cache,
+                              INotificationService notificationService)
+        {
+            _clientId = clientId;
+            _userAgent = userAgent;
+            _userId = userId;
+            _roomName = roomName;
+            _chatService = service;
+            _repository = repository;
+            _cache = cache;
+            _notificationService = notificationService;
+        }
+
+        public string ParseCommand(string commandString, out string[] args)
+        {
+            var parts = commandString.Substring(1).Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+
+            if (parts.Length == 0)
+            {
+                args = new string[0];
+                return null;
+            }
+
+            args = parts.Skip(1).ToArray();
+            return parts[0];
+        }
+        public bool TryHandleCommand(string command)
+        {
+            command = command.Trim();
+            if (!Regex.IsMatch(command, @"^\/[A-Za-z0-9?]+?"))
+            {
+                return false;
+            }
+
+            string[] args;
+            var commandName = ParseCommand(command, out args);
+            return TryHandleCommand(commandName, args);
+        }
+
+        public bool TryHandleCommand(string commandName, string[] args)
+        {
+            if (String.IsNullOrEmpty(commandName))
+            {
+                return false;
+            }
+
+            commandName = commandName.Trim();
+            if (commandName.StartsWith("/"))
+            {
+                return false;
+            }
+
+            var context = new CommandContext
+            {
+                Cache = _cache,
+                NotificationService = _notificationService,
+                Repository = _repository,
+                Service = _chatService
+            };
+
+            var callerContext = new CallerContext
+            {
+                ClientId = _clientId,
+                UserId = _userId,
+                UserAgent = _userAgent,
+                RoomName = _roomName,
+            };
+
+            ICommand command;
+            try
+            {
+                MatchCommand(commandName, out command);
+            }
+            catch (CommandNotFoundException)
+            {
+                throw new HubException(String.Format(LanguageResources.CommandNotFound, commandName));
+            }
+            catch (CommandAmbiguityException e)
+            {
+                throw new HubException(String.Format(LanguageResources.CommandAmbiguous, commandName, String.Join(", ", e.Ambiguities)));
+            }
+
+            command.Execute(context, callerContext, args);
+
+            return true;
+        }
+
+        private void MatchCommand(string commandName, out ICommand command)
+        {
+            throw new NotImplementedException();
```

Comment:
I msgd on over a couple weeks ago some code to try here, give it a whirl. Then we can get commented blocks below removed.